### PR TITLE
Mccalluc/clean up md

### PR DESCRIPTION
**Pull request by mccalluc**
Remove stub pages and fill in links, as discussed.

(As you can imagine, I am not excited about maintaining a parallel list of assay types. Your call for how we handle this now, and in the future.)

### CODE REVIEW

File: context/app/markdown/docs/metadata.md [hunk 1]
```diff
@@ -2,38 +2,73 @@
 
 HuBMAP metadata documentation and submission specifications
 
-## Donor
```
Comment (ngehlenborg) [inline context/app/markdown/docs/metadata.md:5]: This section should remain until it is clear that this won't be delivered.

File: context/app/markdown/docs/metadata.md [hunk 2]
```diff
@@ -2,38 +2,73 @@
 
 HuBMAP metadata documentation and submission specifications
```
Comment (mccalluc) [inline context/app/markdown/docs/metadata.md:4]:
```suggestion

## Donor

Donor Specification

TODO: Donor specification pending from Jonathan Silverstein.

```

File: context/app/markdown/docs/metadata.md (hunk 1 above)
Comment (mccalluc) [reply context/app/markdown/docs/metadata.md:5]: Reverted. Thanks!

**Comment by mccalluc** [comment]
Brought back the donor section, with a TODO. The separate donor.md is still being removed, but it can be brought back easily.

**Pull request merged**